02-weather-app/weather.js

```diff
@@ -0,0 +1,84 @@
+'use strict';
+const key = '6ad0d5e302ac0d03fcc31a072356981f';
+
+const inputCity = document.querySelector('.input');
+const ul = document.querySelector('.input--list');
+const btnChange = document.getElementById('btn-change');
+
+async function search() {
+  const phrase = inputCity.value;
+  const response = await fetch(
+    `http://api.openweathermap.org/geo/1.0/direct?q=${phrase}&limit=5&appid=${key}`
+  );
+  const data = await response.json();
+  ul.innerHTML = '';
+  for (let i = 0; i < data.length; i++) {
+    const { name, lat, lon, country } = data[i];
+    ul.innerHTML += `<li
+    data-lat="${lat}"
+    data-lon="${lon}"
+    data-name="${name}"
+    > ${name} <span>${country}</span></li>`;
+  }
+}
+
+const debounceSearch = _.debounce(() => {
+  search();
+}, 600);
+
+inputCity.addEventListener('keyup', debounceSearch);
+
+async function showWeather(lat, lon, name) {
```

Comment:
i dont really like this shortnames. Why loose readibility to save few characters? (latitude, longitude, name) is really more elaborative :)